content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: 'Cyclic referential actions with Microsoft SQL Server'
+metaTitle: 'Cyclic referential actions with Microsoft SQL Server'
+metaDescription: 'Circular references can occur between models when each model references another creating a closed loop. Since Microsoft SQL Server does not handle these situations out of the box, learn how to solve this problem.'
+tocDepth: 2
+---
+
+<TopBlock>
+
+Circular references can occur between models when each model references another, creating a closed loop. Most SQL databases handle these references out of the box. 
+
+With Microsoft SQL Server, however, if the [referential actions](./) on the relation field is set to something other than `NoAction`, the data model validator will throw an error the cyclic relation.
+
+This is dependent on whether or not the underlying scalar fields are set to be optional or required. If no referential action is specified on the relation, the default values are used:
```

Comment:
link "scalar fields" to our explanation.

Reply:
This sentence is weird to me.

This = If the field is set to NoAction or something different? Or if an error is thrown or not?

Reply:
How about this instead:

> If no referential action is specified on the relation, the default values are used. Knowing which referential action to add is dependent on whether or not the underlying [scalar fields](../../data-model#scalar-fields) are set to be optional or required.

Reply:
Perfect, you understood exactly what I meant 👍